crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -697,6 +764,31 @@ where
 
         Ok(())
     }
+
+    fn add_funding_transaction_to_shadow_channel(
+        &self,
+        user_channel_id: u128,
+        channel_id: [u8; 32],
+    ) -> Result<()> {
+        let user_channel_id = Uuid::from_u128(user_channel_id).to_string();
+        let channel_details = self
+            .channel_manager
+            .get_channel_details(&channel_id)
+            .ok_or(anyhow!(
+                "Failed to get channel details by channel_id {}",
+                hex::encode(channel_id)
+            ))?;
+        if let Some(mut channel) = self.storage.get_channel(&user_channel_id)? {
+            channel.channel_state = ChannelState::Open;
+            channel.funding_txid = channel_details.funding_txo.map(|txo| txo.txid.to_string());
+            channel.channel_id = Some(hex::encode(channel_id));
+            channel.updated_at = OffsetDateTime::now_utc();
```

Comment:
🔧 Might also want to encapsulate this transition in `Channel`.

Reply:
encapsulated into the channel impl.